### PR TITLE
Add support for Mythic Beasts Dynamic DNS

### PR DESCRIPTION
**Pull request by rrthomas**
This is called `mythicdyn`, to differentiate from their more general zone
file editing, support for which could be added in future (it works similarly
to Gandi’s).

This patch also adds support for error code 403, which Mythic uses, and
improves the error message for 401.

### CODE REVIEW

**Comment by wimpunk** [comment]
Looks like some tests are failing on your patch. Can you verify what's going on there?

**Comment by rrthomas** [comment]
@wimpunk, it's not getting as far as running the tests, the CI script is failing on some versions of CentOS (it's failing to install dependencies).

**Comment by SuperSandro2000** [comment]
Please rebase the PR properly.

**Comment by rrthomas** [comment]
@SuperSandro2000 apologies, I should not have pushed the previous rebase, that was just for testing purposes. I have rebased on master again, so the PR now contains just the Mythic support, as before.

**Comment by rrthomas** [comment]
Ping! I believe this is ready to go.

**Comment by SuperSandro2000** [comment]
Can you please fix the conflict?

**Comment by rrthomas** [comment]
> Can you please fix the conflict?

Sure, done!

**Comment by SuperSandro2000** [comment]
Closing due to inactivity. Feel free to comment and I can reopen the PR.

**Comment by rrthomas** [comment]
I'm not sure why this was closed. I rebased as requested, as far as I was concerned the PR was ready. Happy to update it further if required.

**Comment by SuperSandro2000** [comment]
The PR has merge conflicts which must be fixed before it can be merged.

**Comment by rrthomas** [comment]
@SuperSandro2000 thanks for the notification; I've rebased the PR on current `develop`.

File: ddclient.in [hunk 1]
```diff
@@ -5637,6 +5649,83 @@ sub nic_googledomains_update {
...
+        my $reply = geturl(
+            proxy     => opt('proxy'),
+            url       => "https://ipv$ipversion.$config{$h}{'server'}/dns/v2/dynamic/$h",
+            method    => 'POST',
+            login     => $config{$h}{'login'},
+            password  => $config{$h}{'password'},
+            ipversion => $ipversion,
+        );
+        unless ($reply) {
+            failed("%s -- Could not connect to %s.", $h, $config{$h}{'server'});
```
Comment (SuperSandro2000) [inline ddclient.in:5713]:
```suggestion
            failed("Updating service %s failed: %s", $h, $config{$h}{'server'});
```

File: ddclient.in [hunk 2]
```diff
@@ -5637,6 +5649,83 @@ sub nic_googledomains_update {
...
+######################################################################
+## nic_mythicdyn_update
+######################################################################
+sub nic_mythicdyn_update {
+    debug("\nnic_mythicdyn_update --------------------");
+
+    # Update each set configured host.
+    foreach my $h (@_) {
+        info("%s -- Setting IP address.", $h);
+        verbose("UPDATE:", "updating %s", $h);
```
Comment (SuperSandro2000) [inline ddclient.in:5700]:
```suggestion
```
This is duplicated

**Review by SuperSandro2000 — COMMENTED** [review]
sorry about the delay. ddclient is not a priority for me currently

File: ddclient.in (hunk 2 above)
Comment (rrthomas) [reply ddclient.in:5700]: I'm not the expert here, but the general format seems to follow other backends, it's just that the "UPDATE" line is easy to parse out (I presume). For example, in lines 5493-4, the `nic_godaddy_update` subroutine outputs duplicate information on two successive lines, one a call to `info` and one to `verbose`.

**Comment by rrthomas** [comment]
@SuperSandro2000 Thanks for your most recent review. I have accepted one change, queried another, and also rebased on current master.

File: ddclient.in (hunk 2 above)
Comment (SuperSandro2000) [reply ddclient.in:5805]: There is a general inconsistency in the entire code how provider are handled.

**Comment by SuperSandro2000** [comment]
There is a merge conflict, please rebease

File: ddclient.in (hunk 2 above)
Comment (rrthomas) [reply ddclient.in]: Fine, I've removed this line.

**Comment by rrthomas** [comment]
I've rebased.

**Pull request merged**